File: finance-api/src/main/java/com/financesystem/finance_api/modules/tenant/accounts/application/dto/CreateAccountRequest.java
package com.financesystem.finance_api.modules.tenant.accounts.application.dto;

import com.financesystem.finance_api.modules.tenant.accounts.domain.model.AccountName;
import com.financesystem.finance_api.modules.tenant.accounts.domain.model.AccountType;
import com.financesystem.finance_api.modules.tenant.accounts.domain.model.CurrencyCode;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.UUID;

public record CreateAccountRequest(
        @NotNull
        UUID userId,

        @NotNull
        AccountName accountName,

        @Size(max = 100)
        String customAlias,

        @NotNull
        AccountType accountType,

        @NotNull
        CurrencyCode currency,

        Boolean primary
) {
}

File: finance_mobile/lib/infrastructure/models/create_account_request.dart
class CreateAccountRequest {
  final String accountName;
  final String accountType;
  final String currency;
  final String customAlias;

  CreateAccountRequest({
    required this.accountName,
    required this.accountType,
    required this.currency,
    required this.customAlias,
  });

  Map<String, dynamic> toJson() => {
    'accountName': accountName,
    'accountType': accountType,
    'currency': currency,
    'customAlias': customAlias,
  };
}
